packages/back-office-subscribers/nsip-service-user/index.js

```diff
@@ -0,0 +1,71 @@
+const { prismaClient } = require('../lib/prisma');
+module.exports = async (context, message) => {
+	context.log(`invoking nsip-service-user function`);
+	const serviceUserId = message.id;
+
+	if (!serviceUserId) {
+		context.log(`skipping update as serviceUserId is missing`);
+		return;
+	}
+
+	return await prismaClient.$transaction(async (tx) => {
+		const existingServiceUser = await tx.serviceUser.findUnique({
+			where: {
+				serviceUserId
+			}
+		});
+
+		const shouldUpdate =
+			!existingServiceUser ||
+			new Date(context.bindingData.enqueuedTimeUtc) >
+				new Date(existingServiceUser.modifiedAt.toUTCString());
+
+		if (shouldUpdate) {
+			const serviceUserType = message.serviceUserType;
+
+			const serviceUser = mapMessageToServiceUser(message);
+			await tx.serviceUser.upsert({
+				where: {
+					serviceUserId
+				},
+				update: serviceUser,
+				create: serviceUser
+			});
+
+			context.log(
+				`upserted service user of type ${serviceUserType} with serviceUserId: ${serviceUserId}`
+			);
+		} else {
+			context.log(`skipping update of service user with serviceUserId: ${serviceUserId}`);
+		}
+	});
+};
+
+const mapMessageToServiceUser = (message) => {
+	const serviceUserType = message.serviceUserType;
+	if (!['Representative', 'Represented', 'Applicant'].includes(serviceUserType)) {
+		throw new Error(`Invalid serviceUserType: ${serviceUserType}`);
+	}
+	if (serviceUserType === 'Representative' || serviceUserType === 'Represented') {
+		return {
+			serviceUserId: message.id,
+			firstName: message?.firstName,
```

Comment:
```suggestion
			firstName: message.firstName,
```

is the `?` needed (on this line and all below)? I think `message` should always be defined

Reply:
Good spot thanks - not sure why I had that